frappe/database/database.py

```diff
@@ -158,6 +158,10 @@ def sql(self, query, values=(), as_dict = 0, as_list = 0, formatted = 0,
 				# only for mariadb
 				frappe.errprint('Syntax error in query:')
 				frappe.errprint(query)
+			elif self.is_deadlocked(e) or self.is_timedout(e):
+				err_msg = _('There was a problem accessing required resources to complete the transaction. Please retry.')
+				exception = frappe.QueryTimeout if self.is_timedout(e) else frappe.QueryDeadlock
+				frappe.throw(msg=err_msg, title=_('Request Timeout'), exc=exception)
```

Comment:
Let's change the title accordingly too right? like "Deadlock occurred" when a deadlock occurs.

Perhaps different `elif` blocks would do the trick here.